silkrpc/commands/eth_api.cpp

```diff
@@ -296,12 +296,40 @@ asio::awaitable<void> EthereumRpcApi::handle_eth_get_uncle_by_block_hash_and_ind
 
 // https://eth.wiki/json-rpc/API#eth_getunclebyblocknumberandindex
 asio::awaitable<void> EthereumRpcApi::handle_eth_get_uncle_by_block_number_and_index(const nlohmann::json& request, nlohmann::json& reply) {
+    auto params = request["params"];
+    if (params.size() != 2) {
+        auto error_msg = "invalid eth_getUncleByBlockNumberAndIndex params: " + params.dump();
+        SILKRPC_ERROR << error_msg << "\n";
+        reply = make_json_error(request["id"], 100, error_msg);
+        co_return;
+    }
+    const auto block_id = params[0].get<std::string>();
+    const auto index = params[1].get<std::string>();
+    SILKRPC_DEBUG << "block_id: " << block_id << " index: " << index << "\n";
+
     auto tx = co_await database_->begin();
 
     try {
         ethdb::TransactionDatabase tx_database{*tx};
 
-        reply = make_json_content(request["id"],  "0x" + to_hex_no_leading_zeros(0));
+        const auto block_number = co_await core::get_block_number(block_id, tx_database);
+        const auto block_with_hash = co_await core::rawdb::read_block_by_number(tx_database, block_number);
+        const auto ommers = block_with_hash.block.ommers;
+
+        auto idx = std::stoul(index);
+        if (idx >= ommers.size()) {
+            reply = make_json_content(request["id"], nullptr);
+            co_return;
+        }
+        const auto total_difficulty = co_await core::rawdb::read_total_difficulty(tx_database, block_with_hash.hash, block_number);
+        auto uncle = ommers[idx];
+
+        const silkworm::BlockBody block_body{};
+        const evmc::bytes32 hash {};
+        silkworm::BlockWithHash block_hash{{block_body, uncle}, hash};
```

Comment:
Some improvements here:
1) naming seems ambiguous (it's not an hash) and too similar with line 316
2) it should be more clear that we're creating an empty block to hold the uncle block header
3) `hash` is empty here, whilst it should be `uncle.hash()`

In short I suggest to replace line 329 with:

`silkworm::BlockWithHash uncle_block_with_hash{{{}, uncle}, uncle.hash()};`